src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/kameugne2014/PropCumulative.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature.kameugne2014;
+
+import org.chocosolver.solver.constraints.nary.cumulative.literature.CumulativeFilter;
+import org.chocosolver.solver.constraints.nary.cumulative.literature.vilim2009.ThetaLambdaTree;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+
+import java.util.Comparator;
+
+/**
+ * Cumulative constraint filtering algorithms described in the following paper :
+ * Kameugne, R., Fotso, L.P.: A cumulative not-first/not-last filtering algorithm in O(n2 log(n)). Indian Journal of Pure and Applied Mathematics 44(1), pages 95-115 (2013). https://doi.org/10.1007/s13226-013-0005-z
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public class PropCumulative extends CumulativeFilter {
```

Comment:
The class name should not be so generic, there are many other `PropCumulative` classes, that may be confusing.